beprof/curve.py

```diff
@@ -144,6 +147,70 @@ def rebinned(self, step=0.1, fixp=0):
         domain = [fixp + n * step for n in range(int(count_start), int(count_stop)+1)]
         return self.change_domain(domain)
 
+    def evaluate_at_x(self, arg, defval=0):
+        '''
+        Returns Y value at arg of self. Arg can be a scalar, but also might be np.array or other iterable
+        (like list). If domain of self is not wide enought to interpolate the value of Y, method will return
+        defval for those arugments instead.
+
+        Check the interpolation when arg in domain of self:
+        >>> Curve([[0, 0], [2, 2], [4, 4]]).evaluate_at_x([1, 2 ,3])
+        [ 1.  2.  3.]
+        
+        Check if behavior of the method is correct when arg partly outside the domain:
+        >>> Curve([[0, 0], [2, 2], [4, 4]]).evaluate_at_x([-1, 1, 2 ,3, 5], 100)
+        [ 100.    1.    2.    3.  100.]
+
+        :param arg: x-value to calculate Y (may be an array or list as well)
+        :param defval: default value to return if can't interpolate value at arg
+        :return: np.array of Y-values at arg. If arg is a scalar, will return scalar as well
+        '''
+        y = np.interp(arg, self.x, self.y, left=defval, right=defval)
+        return y
+
+    def subtract(self, curve2, newobj=False,):
```

Comment:
Nothing I didn't notice it. I wasn't even aware the program will run with such a thing.